File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/Document.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

import java.util.Objects;

public class Document {

    private String name;
    private String url;

    /**
     * @return name
     */
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public Document withName(String name) {
        this.name = name;
        return this;
    }

    /**
     * @return url
     */
    public String getUrl() {
        return url;
    }
    public void setUrl(String url) {
        this.url = url;
    }
    public Document withUrl(String url) {
        this.url = url;
        return this;
    }

    @Override
    public boolean equals(java.lang.Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Document document = (Document) o;
        return Objects.equals(this.name, document.name)
                && Objects.equals(this.url, document.url);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, url);
    }

    @Override
    public String toString() {
        return "class Document {\n"
                + "        name: " + toIndentedString(name) + "\n"
                + "        url: " + toIndentedString(url) + "\n"
                + "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
     */
    private String toIndentedString(java.lang.Object o) {
        return o == null
                ? "null"
                : o.toString().replace("\n", "\n        ");
    }
}

File: upl-purchase-sdk-js/src/unzer-paylater/model/Document.js
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

const ModelHelper = require("../util/ModelHelper");

class Document {
    /**
     * @returns { String }
     */
    getName() {
        return this.name;
    }
    /**
     * @param { String } name
     */
    setName(name) {
        this.name = ModelHelper.validatePrimitive(name, "string");
    }
    /**
     * @param { String } val
     */
    withName(val) {
        this.setName(val);
        return this;
    }

    /**
     * @returns { String }
     */
    getUrl() {
        return this.url;
    }
    /**
     * @param { String } url
     */
    setUrl(url) {
        this.url = ModelHelper.validatePrimitive(url, "string");
    }
    /**
     * @param { String } val
     */
    withUrl(val) {
        this.setUrl(val);
        return this;
    }

    /**
     * @returns { Document }
     */
    static constructFromObject(data) {
        if (!data) return undefined;
        return new Document()
            .withName(ModelHelper.convertToType(data["name"], String))
            .withUrl(ModelHelper.convertToType(data["url"], String));
    }
}

module.exports = Document;
